x-pack/plugins/ml/public/application/explorer/swimlane_annotation_container.tsx

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useEffect } from 'react';
+import d3 from 'd3';
+import { scaleTime } from 'd3-scale';
+import { i18n } from '@kbn/i18n';
+import { formatHumanReadableDateTimeSeconds } from '../../../common/util/date_utils';
+import { AnnotationsTable } from '../../../common/types/annotations';
+import { ChartTooltipService } from '../components/chart_tooltip';
+
+const ANNOTATION_HEIGHT = 12;
+export const Y_AXIS_LABEL_WIDTH = 170;
+export const Y_AXIS_LABEL_PADDING = 8;
+export const Y_AXIS_LABEL_FONT_COLOR = '#6a717d';
+
+interface SwimlaneAnnotationContainerProps {
+  chartWidth: number;
+  domain: {
+    min: number;
+    max: number;
+  };
+  annotationsData?: AnnotationsTable['annotationsData'];
+  tooltipService: ChartTooltipService;
+}
+
+export const SwimlaneAnnotationContainer: FC<SwimlaneAnnotationContainerProps> = ({
+  chartWidth,
+  domain,
+  annotationsData,
+  tooltipService,
+}) => {
+  const canvasRef = React.useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    if (canvasRef.current !== null && Array.isArray(annotationsData)) {
+      const chartElement = d3.select(canvasRef.current);
+      chartElement.selectAll('*').remove();
+
+      const startingXPos = Y_AXIS_LABEL_WIDTH + 2 * Y_AXIS_LABEL_PADDING;
+      const endingXPos = startingXPos + chartWidth - Y_AXIS_LABEL_PADDING;
+
+      const svg = chartElement
+        .append('svg')
+        .attr('width', '100%')
+        .attr('height', ANNOTATION_HEIGHT);
+
+      const xScale = scaleTime().domain([domain.min, domain.max]).range([startingXPos, endingXPos]);
+
+      // Add Annotation y axis label
+      svg
+        .append('text')
+        .attr('text-anchor', 'end')
+        .attr('class', 'swimlaneAnnotationLabel')
+        .text(
+          i18n.translate('xpack.ml.explorer.swimlaneAnnotationLabel', {
+            defaultMessage: 'Annotations',
+          })
+        )
+        .attr('x', Y_AXIS_LABEL_WIDTH + Y_AXIS_LABEL_PADDING)
+        .attr('y', ANNOTATION_HEIGHT)
+        .style('fill', Y_AXIS_LABEL_FONT_COLOR)
+        .style('font-size', '12px');
+
+      // Add border
+      svg
+        .append('rect')
+        .attr('x', startingXPos)
+        .attr('y', 0)
+        .attr('height', ANNOTATION_HEIGHT)
+        .attr('width', endingXPos - startingXPos)
+        .style('stroke', '#cccccc')
+        .style('fill', 'none')
+        .style('stroke-width', 1);
+
+      // Add annotation marker
+      annotationsData.forEach((d) => {
```

Comment:
Can the markers be given the same rounded borders for consistency with the Single Metric Viewer?

Anomaly Explorer:
![image](https://user-images.githubusercontent.com/7405507/115221680-d5446280-a101-11eb-9bae-60e51b94d2b1.png)

SMV:
![image](https://user-images.githubusercontent.com/7405507/115221630-c9f13700-a101-11eb-9709-2e9c0ce7134e.png)

Reply:
Updated the styling here [`229b4fa` (#97202)](https://github.com/elastic/kibana/pull/97202/commits/229b4fa7d8a4e1d082dc388f912b353932288e80)

Anomaly Explorer:
<img width="1045" alt="Screen Shot 2021-04-20 at 09 45 11" src="https://user-images.githubusercontent.com/43350163/115416235-354c0f00-a1bd-11eb-9d27-1dd6ab7bf617.png">

SMV:
<img width="1045" alt="Screen Shot 2021-04-20 at 09 45 18" src="https://user-images.githubusercontent.com/43350163/115416263-3c731d00-a1bd-11eb-8357-f3a713b9f92d.png">